mtls-northbound/README.md

```diff
@@ -0,0 +1,603 @@
+gcloud privateca roots create ${ROOT} \
+    --pool=${POOL} \
+    --subject="CN=partner1-ca, O=Partner 1" \
+    --location=us-east1 \
+    --auto-enable
+```
+
+## Create Trust Configuration
+The trust configuration can be used by multiple Server TLS Policies (e.g. lenient, strict).
+
+First, get the private CA root certificate.
+```
+gcloud privateca roots describe ${ROOT} \
+   --pool=${POOL} \
+   --location=us-east1 \
+   --format='value(pemCaCertificates)' > ${ROOT}.cert
+```
+Then format to remove newlines and create the trust config yaml file. 
+```
+export ROOT_PEM=$(cat ${ROOT}.cert | sed 's/^[ ]*//g' | tr '\n' $ | sed 's/\$/\\n/g')
+
+cat << EOF > ${TRUST_CONFIG}.yaml
+name: ${TRUST_CONFIG}
+trustStores:
+- trustAnchors:
+   - pemCertificate: "${ROOT_PEM?}"
+EOF
+```
+Finally create the trust config.
+```
+gcloud beta certificate-manager trust-configs import ${TRUST_CONFIG} --source=${TRUST_CONFIG}.yaml
+```
+
+## Create Server TLS Policies
+Create policy representations for "lenient" and "strict", 
+then you can easily switch between them by updating the Target HTTPS Proxy.
+
+### Lenient
+Create the lenient policy yaml file.
+```
+cat << EOF > ${ROOT}-lenient.yaml
+name: ${ROOT}-lenient
+mtlsPolicy:
+  clientValidationMode: ALLOW_INVALID_OR_MISSING_CLIENT_CERT
+  clientValidationTrustConfig: projects/${PROJECT}/locations/global/trustConfigs/${TRUST_CONFIG}
+EOF
+```
+Create the lenient policy.
+```
+gcloud beta network-security server-tls-policies import ${ROOT}-lenient \
+    --source=${ROOT}-lenient.yaml \
+    --location=global
+```
+
+### Strict
+Create the strict policy yaml file.
+```
+cat << EOF > ${ROOT}-strict.yaml
+name: ${ROOT}-strict
+mtlsPolicy:
+  clientValidationMode: REJECT_INVALID
+  clientValidationTrustConfig: projects/${PROJECT}/locations/global/trustConfigs/${TRUST_CONFIG}
+EOF
+```
+Create the strict policy.
+```
+gcloud beta network-security server-tls-policies import ${ROOT}-strict \
+    --source=${ROOT}-strict.yaml \
+    --location=global
+```
+
+## Prepare Configuration files to Update the GLB
+The steps are the same for any Application GLB, just find the Target HTTPS Proxy for the GLB and update.
+You can easily switch between the TLS policies by creating representations for each configuration.
+
+Find the Target HTTPS Proxy for your GLB.
+```
+gcloud compute target-https-proxies list
+                                SSL_CERTIFICATES            URL_MAP              REGION    CERTIFICATE_MAP
+apigee-proxy-https-proxy        gm-xapi-kurtkanaskie-net    apigee-proxy-url-map
+apigee-proxy-modern-https-proxy gm-m-xapi-kurtkanaskie-net  apigee-proxy-modern-url-map
+```
+
+### Create Target HTTPS Proxy configurations
+We'll create multiple configurations so we can easily switch between them, one for "none", "lenient" and "strict".
+
+Export the Target HTTPS Proxy configuration and store in a "none" file.
+This allows us to remove the Server TLS Policy later.
+```
+export TARGET_PROXY=apigee-proxy-https-proxy
```

Comment:
We should make it more clear they need to replace with the name of the target proxy from their project

Reply:
OK